src/sync.cpp

```diff
@@ -27,10 +28,9 @@
 #if !defined(HAVE_THREAD_LOCAL)
 static_assert(false, "thread_local is not supported");
 #endif
-void PrintLockContention(const char* pszName, const char* pszFile, int nLine)
+void LockContention(const char* pszName, const char* pszFile, int nLine)
 {
-    LogPrint(BCLog::LOCK, "LOCKCONTENTION: %s\n", pszName);
-    LogPrint(BCLog::LOCK, "Locker: %s:%d\n", pszFile, nLine);
+    LOG_TIME_MICROS_WITH_CATEGORY(strprintf("%s, %s:%d", pszName, pszFile, nLine), BCLog::LOCK);
```

Comment:
I think `LockContention` doesn't do what it is supposed to do. At least it doesn't log the time that the contention actually takes: the macro creates an `BCLog::Timer` instance where constructor measures start time, and destructor prints finished. But since all that is done inside the `LockContention` function, this happens instantly and doesn't measure the actual lock contention.

I've verified that in https://github.com/bitcoin/bitcoin/pull/21006 because I wanted to see how long the contentions are, but it always prints 0µs or 1µs. When I move the macro into `sync.h` and replace 

```cpp
LockContention(pszName, pszFile, nLine); // log the contention`
```
with
```cpp
LOG_TIME_MICROS_WITH_CATEGORY(strprintf("%s, %s:%d", pszName, pszFile, nLine), BCLog::LOCK);
```
I get reasonable contention times, like so:

```
2021-09-06T17:18:23Z Enter: cs_main, rest.cpp:281 completed (65406μs)
2021-09-06T17:18:23Z Enter: cs_main, rest.cpp:281 completed (65331μs)
2021-09-06T17:18:23Z Enter: cs_main, rest.cpp:281 completed (65074μs)
2021-09-06T17:18:23Z Enter: cs_main, rest.cpp:281 completed (65103μs)
2021-09-06T17:18:23Z Enter: cs_main, net_processing.cpp:4051 completed (67687μs)
```

Reply:
Interesting. Right now on mainnet with a non-debug clang 13 build running since 6 hours (2 cores x 2.50GHz), I see varying durations of 1-50 usec, most often 2-30.  I suspect your machine is much faster than mine. Will try your suggestion.

Reply:
@martinus This does seem better, now seeing durations of 1 to several million usec (mainnet, non-debug build). Thanks for reporting. Proposed fix in #22904.